src/Microsoft.OData.Core/ODataBatchReader.cs

```diff
@@ -204,440 +233,82 @@ void IODataBatchOperationListener.BatchOperationContentStreamDisposed()
         }
 
         /// <summary>
-        /// Returns the next state of the batch reader after an end boundary has been found.
+        /// Increases the size of the current batch message; throws if the allowed limit is exceeded.
         /// </summary>
-        /// <returns>The next state of the batch reader.</returns>
-        private ODataBatchReaderState GetEndBoundaryState()
+        protected void IncreaseBatchSize()

```

Comment:
>IncreaseBatchSize [](start = 23, length = 17)

Can IncreaseBatchSize and IncreaseChangesetSize be made private, and called in CreateOperationRequestMessage in the base class?

Reply:
They both are invoked only from ReadImplementation().

---
In reply to: [141146669](https://github.com/OData/odata.net/pull/960#discussion_r141146669) [](ancestors = 141146669)

Reply:
As long as these are private and only called from the base class, I'm good.  There just shouldn't be a reason for the derived class to call these methods.

---
In reply to: [142289160](https://github.com/OData/odata.net/pull/960#discussion_r142289160) [](ancestors = 142289160,141146669)